tests/test_model_bad_units.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+import os
+
+from cellmlmanip import parser
+from cellmlmanip import units
+
+
+class TestModelBadUnits:
+    @pytest.fixture(scope="class")
+    def parser_instance(self):
+        """Parses example CellML and returns model"""
+        example_cellml = os.path.join(
+            os.path.dirname(__file__), "cellml_files", "simple_model_invalid_units.cellml"
+        )
+        p = parser.Parser(example_cellml)
+        return p
+
+    @pytest.fixture(scope="class")
+    def model(self, parser_instance):
+        model = parser_instance.parse()
+        return model
+
+    def test_units(self, parser_instance, model):
+        model.get_equation_graph(True)  # set up the graph - it is not automatic
+        symbol_a = model.get_symbol_by_cmeta_id("a")
+        symbol_b = model.get_symbol_by_cmeta_id("b")
+        equation = model.get_equations_for([symbol_b])
+        assert len(equation) == 2
+        assert equation[0].lhs == symbol_a
+        assert equation[0].rhs == 2.0
+        assert model.units.summarise_units(equation[0].lhs) == 'ms'
+        # cellml file states a (ms) = 1 (ms) + 1 (second)
+        # this does not raise and error
```

Comment:
Yeah, that's actually consistent with the spec bizarrely (although pycml will flag it up) - processing s/w is only required to check connections. And iirc (@MichaelClerx will know for sure) in CellML 2 s/w *shouldn't* check inside equations!

Reply:
I added an new issue #90 where we can discuss what we did with this.